clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaConsumerService.java

```diff
@@ -76,6 +78,9 @@ public abstract class KafkaConsumerService extends AbstractKafkaConsumerService
   protected final Map<PubSubTopic, Map<PubSubTopicPartition, SharedKafkaConsumer>> versionTopicToTopicPartitionToConsumer =
       new VeniceConcurrentHashMap<>();
 
+  // This per consumer lock is for protecting the partition unsubscription and data receiver setting operations.
```

Comment:
Nit point, maybe one sentence to briefly remind future readers that using the `SharedKafkaConsumer's` intrinsic lock could cause the race and redirect them to the PR's description for details.

Reply:
Add one sentence as suggested